lib/MusicBrainz/Server/Data/EditNote.pm

```diff
@@ -73,10 +77,17 @@ sub insert
 sub add_note
 {
     my ($self, $edit_id, $note_hash) = @_;
+
+    my $edit = $self->c->model('Edit')->get_by_id($edit_id)
+               or die "Edit $edit_id does not exist!";
+    my $note_editor_id = $note_hash->{editor_id};
+    my $is_modbot = $note_editor_id == $EDITOR_MODBOT;
+    my $note_editor = $self->c->model('Editor')->get_by_id($note_editor_id);
+    return unless $is_modbot || $edit->editor_may_add_note($note_editor);
```

Comment:
This PR seems fine, but yes, ideally this is already prevented at the controller level. A check in the data layer is useful, but I would explicitly error, because it indicates that the controller (or other script) is doing something wrong. (In the suggestion below, I also skipped loading/checking `$note_editor` at all if we already know it's ModBot.)
```suggestion
    unless ($note_editor_id == $EDITOR_MODBOT) {
        my $note_editor = $self->c->model('Editor')->get_by_id($note_editor_id);
        $edit->editor_may_add_note($note_editor)
            or die "Editor $note_editor_id may not add a note to edit $edit_id";
    }
```
```